actions/python-setup-poetry/action.yaml

```diff
@@ -18,35 +18,28 @@ inputs:
 runs:
   using: "composite"
   steps:
-    - name: Set up Python ${{ inputs.python-version }}
-      uses: actions/setup-python@v2
-      with:
-        python-version: ${{ inputs.python-version }}
-
-    - name: Get full Python version
-      id: full-python-version
-      run: echo ::set-output name=version::$(python -c "import sys; print('-'.join(str(v) for v in sys.version_info))")
-      shell: bash
-
-    - name: Bootstrap poetry
+    - name: Install Poetry ${{ inputs.poetry-version }}
+      run: pipx install poetry==${{ inputs.poetry-version }}
       shell: bash
-      run: |
-        python -m ensurepip
-        python -m pip install --upgrade pip
-        python -m pip install poetry==${{ inputs.poetry-version }}
 
     - name: Configure poetry
-      run: python -m poetry config virtualenvs.in-project true
+      run: |
+        poetry config virtualenvs.in-project true
+        poetry config virtualenvs.path ${{ inputs.working-directory }}/.venv
       shell: bash
 
-    - name: Set up cache
-      uses: actions/cache@v2
-      id: cache
+    - name: Set up Python ${{ inputs.python-version }}
+      uses: actions/setup-python@v4
       with:
-        path: ${{ inputs.working-directory }}/.venv
-        key: venv-${{ runner.os }}-${{ steps.full-python-version.outputs.version }}-${{ hashFiles('${{ inputs.working-directory }}/poetry.lock') }}
+        python-version: ${{ inputs.python-version }}
+        cache: poetry
+        cache-dependency-path: ${{ inputs.working-directory }}/poetry.lock
 
-    - name: Ensure cache is healthy
-      if: steps.cache.outputs.cache-hit == 'true'
-      run: timeout 10s python -m poetry run pip --version || rm -rf ${{ inputs.working-directory }}/.venv
+    - run: poetry env use ${{ inputs.python-version }}
+      shell: bash
+      working-directory: ${{ inputs.working-directory }}
+
+    - name: Get full Python version
+      id: full-python-version
+      run: echo ::set-output name=version::$(python -c "import sys; print('-'.join(str(v) for v in sys.version_info))")
```

Comment:
What is the purpose of this step? To output the python version? Do we still need it?
I noticed that you are not using this output anywhere. It is also not properly defined on top of the `action.yaml` file like this:
```yaml
outputs:
  version:
    description: "The full python version."
    value: ${{ steps.full-python-version.outputs.version }}
```
I would also recommend changing the name to something more concrete like `full-python-version`. Please also add it in the README.

Reply:
Wasn't sure either so I left it there. I am happy to remove it though, if it's not needed.